libpod/in_memory_state.go

```diff
@@ -112,9 +143,18 @@ func (s *InMemoryState) HasContainer(id string) (bool, error) {
 		return false, ErrEmptyID
 	}
 
-	_, ok := s.containers[id]
+	ctr, ok := s.containers[id]
+	if ok {
```

Comment:
I would prefer
if !ok {
return false nil
}
Then you can bring the big block in a level.